src/main/java/com/mongodb/kafka/connect/sink/writemodel/strategy/UpdateOneTimestampsStrategy.java

```diff
@@ -48,10 +49,15 @@
                     new DataException(
                         "Could not build the WriteModel,the value document was missing unexpectedly"));
 
-    BsonDateTime dateTime = new BsonDateTime(Instant.now().toEpochMilli());
+    BsonValue idValue = vd.get(ID_FIELD);
+    if (idValue == null) {
+      throw new DataException(
+          "Could not build the WriteModel,the `_id` field was missing unexpectedly");
+    }
 
+    BsonDateTime dateTime = new BsonDateTime(Instant.now().toEpochMilli());
```

Comment:
If only milliseconds precision (a.k.a., resolution - the "fineness" to which a timer can be read) is needed, using `System.currentTimeMillis` instead of `Instant.now().toEpochMilli()` is more efficient because
- a new object is not created;
- a less accurate (a.k.a., more granular; granularity - the average time in for which a timer value stays the same) clock may be used under the hood (this seems to be the case in OpenJDK JDK).

Reply:
Nice - done.